pennylane/templates/subroutines/qrom.py

```diff
@@ -0,0 +1,299 @@
+        initialized to zero.
+
+    """
+
+    def __init__(
+        self, bitstrings, control_wires, target_wires, work_wires, clean=True, id=None
+    ):  # pylint: disable=too-many-arguments
+
+        control_wires = qml.wires.Wires(control_wires)
+        target_wires = qml.wires.Wires(target_wires)
+
+        work_wires = qml.wires.Wires(work_wires) if work_wires else qml.wires.Wires([])
+
+        self.hyperparameters["bitstrings"] = bitstrings
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["target_wires"] = target_wires
+        self.hyperparameters["work_wires"] = work_wires
+        self.hyperparameters["clean"] = clean
+
+        if work_wires:
+            if any(wire in work_wires for wire in control_wires):
+                raise ValueError("Control wires should be different from work wires.")
+
+            if any(wire in work_wires for wire in target_wires):
+                raise ValueError("Target wires should be different from work wires.")
+
+        if any(wire in control_wires for wire in target_wires):
+            raise ValueError("Target wires should be different from control wires.")
+
+        if 2 ** len(control_wires) < len(bitstrings):
+            raise ValueError(
+                f"Not enough control wires ({len(control_wires)}) for the desired number of "
+                + f"bitstrings ({len(bitstrings)}). At least {int(math.ceil(math.log2(len(bitstrings))))} control "
+                + "wires required."
+            )
+
+        if len(bitstrings[0]) != len(target_wires):
+            raise ValueError("Bitstring length must match the number of target wires.")
+
+        all_wires = target_wires + control_wires + work_wires
+        super().__init__(wires=all_wires, id=id)
+
+    def _flatten(self):
+        data = (self.hyperparameters["bitstrings"],)
+        metadata = tuple(
+            (key, value) for key, value in self.hyperparameters.items() if key != "bitstrings"
+        )
+        return data, metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        bitstrings = data[0]
+        hyperparams_dict = dict(metadata)
+        return cls(bitstrings, **hyperparams_dict)
+
+    def __repr__(self):
+        return f"QROM(control_wires={self.control_wires}, target_wires={self.target_wires},  work_wires={self.work_wires}, clean={self.clean})"
+
+    def map_wires(self, wire_map: dict):
+        new_target_wires = [
+            wire_map.get(wire, wire) for wire in self.hyperparameters["target_wires"]
+        ]
+        new_control_wires = [
+            wire_map.get(wire, wire) for wire in self.hyperparameters["control_wires"]
+        ]
+
+        new_work_wires = []
+        if self.hyperparameters["work_wires"]:
+            new_work_wires = [
+                wire_map.get(wire, wire) for wire in self.hyperparameters["work_wires"]
+            ]
```

Comment:
Could do
```suggestion
        new_dict = {key: [wire_map.get(w, w) for w  in self.hyperparameters[key]] for key in ["target_wires", "control_wires", "work_wires"]}
```
and pass the `new_dict` below.